mapfile.c

```diff
@@ -1351,6 +1351,44 @@ int msLoadProjectionStringEPSG(projectionObj *p, const char *value)
   return msLoadProjectionString(p, value);
 }
 
+int msLoadProjectionCodeString(projectionObj* p, const char* value) {
+
+  int num_params = 0;
+
+  // exit if init= is already at the start of the string e.g. from msOGRSpatialRef2ProjectionObj
+  if (strncasecmp(value, "init=", 5) == 0) {
+      return -1;
+  }
+
+  if (!strchr(value, ':')) {
+      return -1;
+  }
+
+  char** papszList = NULL;
+  papszList = msStringSplit(value, ':', &(num_params));
+
+  if (num_params != 2) {
+      return -1;
+  }
+
+  size_t buffer_size = 0;
+  char* init_string = NULL;
+  buffer_size = 5 + strlen(value) + 1;
+  init_string = (char*)msSmallMalloc(buffer_size);
+
+  /* translate into PROJ format. */
+  msStringToLower(papszList[0]);
```

Comment:
On Linux, filename lookups are case sensitive. I would drop the msStringToLower(). I would say it is the responsibility of the user to provide its input in a consistent way with the case of the file.
Or otherwise it should be documented that the filenames of the init files should be lower case

Reply:
Only issue with this is that these codes will likely be read directly from the GetCapabilities for the service, and lots of Mapfiles in the wild (and in msautotest) use capital letters in the metadata: `"wms_srs" "EPSG2:42304 EPSG:42101 EPSG:4269 EPSG:4326"`

These codes would then be passed to the GetMap request as-is which would mean they wouldn't be found on Linux. There are lots of instances of hardcoded `init=epsg:...` in the codebase.

Reply:
just use capital filenames then :-)

init=epsg:XXXX is treated as a special case inside PROJ given all the backward compatibility issues. If there is an "epsg" filename in PROJ_DATA/PROJ_LIB, it is used. Otherwise PROJ will lookup EPSG:XXXX in proj.db

Reply:
True..these are custom authorities so can be named accordingly. Updated.